call-profile/main.go

```diff
@@ -0,0 +1,566 @@
+	"PROFILE_SERVICE_HEALTH":  "PROFILE_SERVICE_HEALTH",
+	"PROFILE_SKIPPED":         "PROFILE_SKIPPED",
+	"PROFILE_DIFF_STORED":     "PROFILE_DIFF_STORED",
+	"STATUS_BLOCKED":          "BLOCKED",
+	"PROFILE_SERVICE_BLOCKED": "PROFILE_SERVICE_BLOCKED",
+	"NOT_APPROVED":            "NOT APPROVED",
+	"PROFILE_SKIPPED_DUE_TO_UNAUTHENTICATED_ACCESS_TO_PROFILE_DATA": "profileSkippedDueToUnAuthenticatedAccessToProfileData",
+	"PROFILE_SKIPPED_DUE_TO_ERROR_IN_GETTING_PROFILE_DATA":          "profileSkippedDueToErrorInGettingProfileData",
+	"SKIPPED_SAME_LAST_REJECTED_DIFF":                               "skippedSameLastRejectedDiff",
+	"SKIPPED_SAME_LAST_PENDING_DIFF":                                "skippedSameLastPendingDiff",
+	"SKIPPED_CURRENT_USER_DATA_SAME_AS_DIFF":                        "skippedCurrentUserDataSameAsDiff",
+	"SKIPPED_OTHER_ERROR":                                           "skippedOtherError",
+	"SKIPPED_VALIDATION_ERROR":                                      "validation error",
+}
+
+/*
+ Setting Firestore Key for development/production
+*/
+func getFirestoreKey() string {
+	if os.Getenv(("environment")) == Constants["ENV_DEVELOPMENT"] {
+		return os.Getenv(Constants["FIRE_STORE_CRED"])
+	} else if os.Getenv(("environment")) == Constants["ENV_PRODUCTION"] {
+		var parameterName string = Constants["FIRE_STORE_CRED"]
+
+		sess := session.Must(session.NewSessionWithOptions(session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+		}))
+
+		svc := ssm.New(sess)
+
+		results, err := svc.GetParameter(&ssm.GetParameterInput{
+			Name: &parameterName,
+		})
+		if err != nil {
+			log.Fatalf(err.Error())
+		}
+
+		return *results.Parameter.Value
+	} else {
+		return ""
+	}
+}
+
+/*
+ Utils
+*/
+
+/*
+ Function to initialize the firestore client
+*/
+func initializeFirestoreClient(ctx context.Context) (*firestore.Client, error) {
+	sa := option.WithCredentialsJSON([]byte(getFirestoreKey()))
+	app, err := firebase.NewApp(ctx, nil, sa)
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := app.Firestore(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return client, nil
+}
+
+func (res Res) Validate() error {
+	return validation.ValidateStruct(&res,
+		validation.Field(&res.FirstName, validation.Required),
+		validation.Field(&res.LastName, validation.Required),
+		validation.Field(&res.Phone, validation.Required, is.Digit),
+		validation.Field(&res.Email, validation.Required, is.Email),
+		validation.Field(&res.YOE, validation.Min(0)),
+		validation.Field(&res.Company, validation.Required),
+		validation.Field(&res.Designation, validation.Required),
+		validation.Field(&res.GithubId, validation.Required),
+		validation.Field(&res.LinkedIn, validation.Required),
+		validation.Field(&res.Website, is.URL))
+}
+
+/*
+ MODELS
+*/
+
+/*
+ Logs the health of the user's service
+*/
+func logHealth(client *firestore.Client, ctx context.Context, userId string, isServiceRunning bool, sessionId string) {
```

Comment:
client.Collection("logs").Add(ctx, newLog) will create a new document in the Firestore collection logs. However, the document will not have a _id field. This is because the Add method does not automatically generate an _id field for the document.

Reply:
Yes and we don't need the _id as we just want to store it. But if one want to access the automatically generated id, they can access it as add method returns a document reference.